contracts/truefi3/interface/IDebtInstrument.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+/**
+ * @title IDebtInstrument
+ * @notice Represents debt in the TrueFi protocol
+ * @dev This is the basic interface for representing Debt in TrueFi
+ * This interface is meant to be extended to support different debt types
+ *
+ * An NFT represents some debt where:
+ *  - Each debt has an owner and a borrower
+ *  - The owner mints and funds a loan
+ *  - The borrower can withdraw funds from the loan
+ *  - The borrower calls repay() to pay back the loan
+ *  - Repayments are withdrawable by the owner
+ *  - Every DebtInstrument has an underlying token
+ *  - Only principal debt is tracked
+ *
+ **/
+interface IDebtInstrument is IERC721 {
+    /// @dev The owner address lends tokens and withdraws repayments
+    function owner() external view returns (address);
```

Comment:
Is `owner` distinct from the NFT owner? That should be a call to the `ownerOf(tokenId)` view from the IERC721 spec.

Reply:
Actually, shouldn't borrower(), token(), principal(), expiry(), etc. all require a `tokenId` arg? Otherwise you would force every NFT contract to be a singleton containing exactly 1 debt, when instead it should be a single NFT contract that can mint multiple NFT IDs (each representing an individual debt).

Reply:
Yes, you are correct. the NFT contract would manage the funds, and each token has its own ID which tracks amounts debt and claimable balance for the borrower/lender